tests/check/signed_video_helpers.c

```diff
@@ -162,17 +178,45 @@ create_signed_nalus_recurrence(const char *str,
 
 /* Creates and initializes a signed video session. */
 signed_video_t *
-get_initialized_signed_video(SignedVideoCodec codec, sign_algo_t algo)
+get_initialized_signed_video(SignedVideoCodec codec, sign_algo_t algo, bool new_private_key)
 {
   signed_video_t *sv = signed_video_create(codec);
   ck_assert(sv);
   char *private_key = NULL;
   size_t private_key_size = 0;
-  SignedVideoReturnCode rc =
-      signed_video_generate_private_key(algo, "./", &private_key, &private_key_size);
-  ck_assert_int_eq(rc, SV_OK);
-  rc = signed_video_set_private_key(sv, algo, private_key, private_key_size);
-  ck_assert_int_eq(rc, SV_OK);
+  SignedVideoReturnCode rc;
+
+  // Generating private keys takes long time. In unit_tests a new private key is only generated if
+  // it's really needed. One RSA key and one ECDSA key is stored globally to handle the scenario.
+  if (algo == SIGN_ALGO_RSA) {
+    if (global_private_key_size_rsa == 0 || new_private_key) {
+      rc = signed_video_generate_private_key(algo, "./", &private_key, &private_key_size);
+      ck_assert_int_eq(rc, SV_OK);
+
+      assert(private_key_size < RSA_PRIVATE_KEY_ALLOC_BYTES);
+      memcpy(global_private_key_rsa, private_key, private_key_size);
+      global_private_key_size_rsa = private_key_size;
+      global_algo_rsa = algo;
+    }
+    rc = signed_video_set_private_key(
+        sv, global_algo_rsa, global_private_key_rsa, global_private_key_size_rsa);
```

Comment:
Since we are inside the if statement, algo is already known. You can use `algo` directly here. Same goes for ECDSA.